go/teams/teams.go

```diff
@@ -390,15 +390,33 @@ func (t *Team) Leave(ctx context.Context, permanent bool) error {
 	return t.postChangeItem(ctx, section, nil, libkb.LinkTypeLeave, nil, nil, payload)
 }
 
+func (t *Team) traverseUpUntil(ctx context.Context, validator func(t *Team) bool) (targetTeam *Team, err error) {
+	targetTeam = t
+	for !validator(targetTeam) && targetTeam.Chain.inner.ParentID != nil {
+		targetID := targetTeam.Chain.inner.ParentID
+		targetTeam, err = GetForTeamManagement(ctx, targetTeam.G(), *targetID)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return
+}
+
 func (t *Team) getAdminPermission(ctx context.Context, required bool) (admin *SCTeamAdmin, err error) {
 	me, err := t.loadMe(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	// TODO -- recursively try parent teams if this one isn't an admin team.
-	// See CORE-5051
-	logPoint := t.Chain.GetAdminUserLogPoint(me.ToUserVersion())
+	uv := me.ToUserVersion()
+	target_team, err := t.traverseUpUntil(ctx, func(s *Team) bool {
+		return s.Chain.GetAdminUserLogPoint(uv) != nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	logPoint := target_team.Chain.GetAdminUserLogPoint(uv)
```

Comment:
I think this will crash if it can't find an admin permission. Probably want to return `nil, nil` if the traversal fails.

Also, nix the `snake_case` `target_team`